teuthology/task/kube_spray.py

```diff
@@ -0,0 +1,196 @@
+        super(kube_setup, self).begin()
+        self.execute_playbook()
+
+    def generate_hosts_file(self):
+        hosts_dict = dict()
+        for group in sorted(self.groups_to_roles.keys()):
+            role_prefix = self.groups_to_roles[group]
+            want = lambda role: role.startswith(role_prefix)
+            for (remote, roles) in self.cluster.only(want).remotes.iteritems():
+                hostname = remote.hostname
+                if group.startswith('master') or group.startswith('node'):
+                    group_name = 'kube-' + group
+                else:
+                    group_name = group
+                if group_name not in hosts_dict:
+                    hosts_dict[group_name] = {}
+
+                if group_name == 'all':
+                    vars = "ip={ip} ansible_host={ip}".format(ip=remote.ip_address)
+                    hosts_dict[group_name][hostname] = hostname + " " + vars
+                else:
+                    hosts_dict[group_name][hostname] = hostname
+
+        hosts_stringio = StringIO()
+        for group in sorted(hosts_dict.keys()):
+            hosts_stringio.write('[%s]\n' % group)
+            for host in sorted(hosts_dict[group].keys()):
+                host_line = hosts_dict[group][host]
+                hosts_stringio.write('%s\n' % host_line)
+            hosts_stringio.write('\n')
+        # Add the final k8s-cluster: group
+        hosts_stringio.write('\n[k8s-cluster:children]\n')
+        hosts_stringio.write('kube-master\nkube-node\n')
+        hosts_stringio.seek(0)
+        self.inventory = self._write_hosts_file(prefix='kube_hosts_',
+                                                content=hosts_stringio.read().strip())
+
+    def _write_hosts_file(self, prefix, content):
+        """
+        Actually write the hosts file
+        """
+        hosts_file = NamedTemporaryFile(prefix=prefix,
+                                        delete=False)
+        hosts_file.write(content)
+        hosts_file.flush()
+        return hosts_file.name
+
+    def run_pre_req(self):
+        """
+        Install required pre-requistes on each node for kubernetes
+        cluster
+        """
+        for remote in self.cluster.remotes.keys():
+            remote.run(args=[
+                    'sudo', 'sytemctl', 'firewalld', 'disable'
+            ])
+            install_package('ansible', remote)
+
+    def execute_playbook(self):
+        """"
+        run kubespray playbook as defined in usage
+        refer https://github.com/kubernetes-sigs/kubespray
+        """
+        args = [
+            'ANSIBLE_STDOUT_CALLBACK=debug',
+            'ansible-playbook', '-vv',
+            '-i', 'inventory/mycluster/hosts.ini',
+            '--become', '--become-user=root', 'cluster.yml'
+        ]
+        (kube_installer,) = self.ctx.cluster.only('Kmaster.0').remotes.iterkeys()
+        self.kube_installer = kube_installer
+        self.run_pre_req()
+        kube_installer.run(
```

Comment:
The is no sense to use so much quotation, and so many line numbers. 
You can use compact form for better readability, like that:
```
kube_installer.run(args='''rm -rf ~/kubespray''', check_status=False)
```
Same for the code below.